File: 프로그래머스/lv0/120818. 옷가게 할인 받기/옷가게 할인 받기.cs
using System;

public class Solution {
    public int solution(int price) {
        
        float discountedPrice = price;
        
        if(price >= 500000)
        {
            discountedPrice *= 0.8f;
        }
        else if(price >= 300000)
        {
            discountedPrice *= 0.9f;
        }
        else if(price >= 100000)
        {
            discountedPrice *= 0.95f;
        }
        
        return (int)Math.Floor(discountedPrice);
    }
}

File: 프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int price) {
    
    if (price < 100000)      return price;
    else if (price < 300000) return price * 0.95f;
    else if (price < 500000) return price * 0.90f;
    else                     return price * 0.80f;
}